dist/js/shepherd.js

```diff
@@ -207,18 +206,23 @@ var Step = (function (_Evented) {
   }, {
     key: 'getAttachTo',
     value: function getAttachTo() {
+      var returnOpts = {};
       var opts = parsePosition(this.options.attachTo) || {};
-      var selector = opts.element;
-
-      if (typeof selector === 'string') {
-        opts.element = document.querySelector(selector);
+      returnOpts.on = opts.on;
+      if (typeof opts.offset !== 'undefined') {
+        returnOpts.offset = opts.offset;
+      }
 
-        if (!opts.element) {
-          throw new Error('The element for this Shepherd step was not found ' + selector);
+      if (typeof opts.element === 'string') {
+        // Can't override the element in user opts reference because we can't
+        // guarantee that the element will exist in the future.
+        returnOpts.element = document.querySelector(opts.element);
+        if (!returnOpts.element) {
+          throw new Error('The element for this Shepherd step was not found ' + opts.element);
```

Comment:
I'm a little concerned that throwing an error may be more aggressive than is necessary.  It might be better to use `console.error` here, to not prevent other js on the page from executing properly.